[MobileMiniBrowser] Settings popover throws exception on iPad simulator

rreno opened the pull request:
#### 1713bed5d0c732d56f06c5947117844dbdc4f765
<pre>
[MobileMiniBrowser] Settings popover throws exception on iPad simulator
<a href="https://bugs.webkit.org/show_bug.cgi?id=247597">https://bugs.webkit.org/show_bug.cgi?id=247597</a>
rdar://102067010

Reviewed by Simon Fraser.

This fixes the presentation of the settings view to follow the popover API and
makes the settings view work with autolayout.

We will now anchor the view to the settings button so that iPad can correctly display
it as a popover.

* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/Base.lproj/Main.storyboard:
* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController showSettings:]):

Canonical link: <a href="https://commits.webkit.org/256436@main">https://commits.webkit.org/256436@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/542ba2209b4c4a9acd1f0939bf01101ef1089067

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95751 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5005 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28793 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105328 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/165634 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99734 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/5084 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/33761 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88132 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/101163 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101412 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3737 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82363 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/30792 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/87510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73621 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39496 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19048 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37189 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/20365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4449 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/41254 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43003 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/65/builds/43221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39616 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
```diff
@@ -139,8 +139,9 @@ - (IBAction)showTabs:(id)sender
 - (IBAction)showSettings:(id)sender
 {
+    UIPopoverPresentationController *presentationController = [self.settingsViewController popoverPresentationController];
+    presentationController.barButtonItem = sender;
```
It's not a good policy to assume that `sender` is a button. If this were ever hooked up to some other UI (e.g. a menu item) that assumption would be invalid.

Better to add an IBOutlet for the button.

rreno committed:
[MobileMiniBrowser] Settings popover throws exception on iPad simulator
https://bugs.webkit.org/show_bug.cgi?id=247597
rdar://102067010

Reviewed by Simon Fraser.

This fixes the presentation of the settings view to follow the popover API and
makes the settings view work with autolayout.

We will now anchor the view to the settings button so that iPad can correctly display
it as a popover.

* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/Base.lproj/Main.storyboard:
* Tools/MobileMiniBrowser/MobileMiniBrowserFramework/WebViewController.m:
(-[WebViewController showSettings:]):

Canonical link: https://commits.webkit.org/256436@main
